shared/elements/gaia_grid/js/grid_dragdrop.js

```diff
@@ -439,6 +466,25 @@
       this.icon.render();
       this.icon.setCoordinates(oldX, oldY);
 
+      if (!doProcessing) {
+        if (this.iconPositionTimeout === null && !this.isScrolling) {
```

Comment:
Do you think there is any way that we can move this logic into its own method? Call sites would then need to call it after calling ```positionIcon```. It might make sense to break out the positioning, and rearranging of icons. Something like:

```
// Immediate processing.
this.updatePosition();
this.rearrangeGrid();

// Delayed processing.
this.updatePosition();
```